python/lsst/resources/http.py

```diff
@@ -34,8 +37,16 @@
 except ImportError:
     import xml.etree.ElementTree as eTree
 
+try:
+    import fsspec
+    from fsspec.spec import AbstractFileSystem
+except ImportError:
+    fsspec = None
+    AbstractFileSystem = type
+
 from urllib.parse import parse_qs
 
+import aiohttp
```

Comment:
I think this will likely need an RFC to formally add it to rubinenv conda environment. I think it comes in through webdav4 at the moment so things will work. Also need to list it in requirements.txt and pyproject.toml (the latter for the http option) -- or else put protections around the import if this is only used with fsspec.

Reply:
(regardless of import protections we still formally need rubinenv to list it since we want to ensure that this code works in production).

Reply:
I think `aiohttp` is already in the environment, since I could find it. In fact, `fsspec` is built on top of it.

Reply:
```bash
$ source /cvmfs/sw.lsst.eu/almalinux-x86_64/lsst_distrib/w_2024_38/loadLSST.sh

$ command -v python
/cvmfs/sw.lsst.eu/almalinux-x86_64/lsst_distrib/w_2024_38/conda/envs/lsst-scipipe-9.0.0-exact/bin/python

$ python -c "import aiohttp"

$ echo $?
0
```

Reply:
To be precise, I am _not_ using `webdav4`. I preferred not to since after 4 weeks I got no response to an issue I submitted to them, so I don't want to depend on that package (see https://github.com/skshetry/webdav4/issues/196)

Reply:
I agree with Fabio -- I don't think the RFC is needed.  We only need `aiohttp` if we have `fsspec`, and having `fsspec` implies `aiohttp`.  The interface Fabio is using requires him to pass an `aiohttp` client to `fsspec`, so if `fsspec` stops using `aiohttp` internally this is already going to break.

However, I think this import should be part of the `try` block for `fsspec`, or lazy-import it in the function below where it is used.  We have some services that install this library via `pip`.  `fsspec` isn't listed as a package dependency so we don't get it by default.

Reply:
✅ Done: import guarded by same `try` block used for `fsspec`

Reply:
What I'm trying to say is that if we explicitly need aiohttp for this code to work the way we want (ie fsspec working on webdav in the production environment) then we need to make sure that rubin-env comes with aiohttp even if internally fsspec drops that dependency.

Reply:
Tim -- this code is guaranteed to not work if fsspec drops that dependency in any case.  The fsspec constructor signature we are calling is:
```
HTTPFileSystem(get_client: aiohttp.ClientSession)
``` 

So if fsspec stops using aiohttp internally, it's going to break no matter what we do with the conda environment.